codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/compression/PerMessageDeflateClientExtensionHandshakerTest.java

```diff
@@ -121,4 +125,61 @@ public void testCustomHandshake() {
+                -68, 61, 29, 40, 89, -112, -73, 25, 35, 120, -105, -67, -32, -43, -70, -84, 120, -55, 69, 43, -124, 106,
+                -92, 18, -110, 114, -50, 111, 25, -3, 10, 17, -75, 13, 127, -84, 106, 90, -66, 84, -75, 84, 53, -89,
+                -75, 92, -3, -40, -61, 119, 49, -117, 30, 49, 68, -59, 88, 74, -119, -34, 1, -83, -7, -48, 124, -124,
+                -23, 16, 88, -118, 121, 54, -53, 1, 44, 32, 81, 19, 25, -115, -43, -32, -64, -67, -120, -110, -101, 121,
+                -2, 2
+        };
+
+        byte[] secondPayload = new byte[] {
+                -86, 86, 42, 46, 77, 78, 78, 45, 6, 26, 83, 82, 84, -102, -86, 3, -28, 38, 21, 39, 23, 101, 38, -91, 2,
+                -51, -51, 47, 74, 73, 45, 114, -54, -49, -49, -10, 49, -78, -118, 112, 10, 9, 13, 118, 1, -102, 84,
+                -108, 90, 88, 10, 116, 27, -56, -84, 124, -112, -13, 16, 26, 116, -108, 18, -117, -46, -127, 6, 69, 99,
+                -45, 24, 91, 91, 11, 0
+        };
+
+        Map<String, String> parameters =  Collections.singletonMap(CLIENT_NO_CONTEXT, null);
+
+        WebSocketClientExtension extension = handshaker.handshakeExtension(
+                new WebSocketExtensionData(PERMESSAGE_DEFLATE_EXTENSION, parameters));
+        assertNotNull(extension);
+
+        EmbeddedChannel decoderChannel = new EmbeddedChannel(extension.newExtensionDecoder());
+        assertTrue(
+                decoderChannel.writeInbound(new TextWebSocketFrame(true, RSV1, Unpooled.copiedBuffer(firstPayload))));
+        TextWebSocketFrame firstFrameDecompressed = decoderChannel.readInbound();
+        assertTrue(
+                decoderChannel.writeInbound(new TextWebSocketFrame(true, RSV1, Unpooled.copiedBuffer(secondPayload))));
+        TextWebSocketFrame secondFrameDecompressed = decoderChannel.readInbound();
+
+        assertNotNull(firstFrameDecompressed);
+        assertNotNull(firstFrameDecompressed.content());
+        assertTrue(firstFrameDecompressed instanceof TextWebSocketFrame);
+        assertEquals(firstFrameDecompressed.text(),
+                     "{\"info\":\"Welcome to the BitMEX Realtime API.\",\"version\"" +
+                     ":\"2018-10-02T22:53:23.000Z\",\"timestamp\":\"2018-10-15T06:43:40.437Z\"," +
+                     "\"docs\":\"https://www.bitmex.com/app/wsAPI\",\"limit\":{\"remaining\":39}}");
+        firstFrameDecompressed.release();
+
+        assertNotNull(secondFrameDecompressed);
+        assertNotNull(secondFrameDecompressed.content());
+        assertTrue(secondFrameDecompressed instanceof TextWebSocketFrame);
+        assertEquals(secondFrameDecompressed.text(),
+                     "{\"success\":true,\"subscribe\":\"orderBookL2:XBTUSD\"," +
+                     "\"request\":{\"op\":\"subscribe\",\"args\":[\"orderBookL2:XBTUSD\"]}}");
+        secondFrameDecompressed.release();
+
+        decoderChannel.finish();
```

Comment:
@langchristian96 can you please assert that this returns false ?

Reply:
Oops, sorry. Forgot to assert this as well. Also added asserts for releases.